l1-contracts/src/UniFiAVSManager.sol

```diff
@@ -212,6 +225,156 @@ contract UniFiAVSManager is UniFiAVSManagerStorage, IUniFiAVSManager, UUPSUpgrad
+        uint256 newValidatorCount = 0;
+        bytes memory delegateKey = $.operators[msg.sender].commitment.delegateKey;
+
+        for (uint256 i = 0; i < paramsArray.length; i++) {
+            ValidatorRegistrationParams memory params = paramsArray[i];
+
+            // Derive the BLS public key hash from pubkeyG1
+            bytes32 blsPubKeyHash = params.blsPubKeyHash;
+
+            ValidatorData storage existingValidator = $.validators[blsPubKeyHash];
+            ValidatorRegistrationData storage validatorRegistrationData = $.validatorRegistrations[blsPubKeyHash];
+
+            // Check if the validator already exists with active status
+            if (existingValidator.index != 0 && existingValidator.registeredUntil == type(uint64).max) {
+                revert ValidatorAlreadyRegistered();
+            }
+
+            if ($.validatorIndexes[params.index] != bytes32(0)) {
+                revert ValidatorIndexAlreadyUsed();
+            }
+
+            // Check if the registration salt has been used before
+            if (validatorRegistrationData.salt == params.salt) {
+                revert SaltAlreadyUsed();
+            }
+
+            // Check if the signature is expired
+            if (block.timestamp > params.expiry) {
+                revert SignatureExpired();
+            }
+
+            // Store the validator data
+            $.validators[blsPubKeyHash] = ValidatorData({
+                eigenPod: address(0), // Not from EigenPod
+                index: params.index, // Store the provided index
+                operator: msg.sender,
+                registeredUntil: type(uint64).max,
+                registeredAfter: uint64(block.number) + $.registerationDelay
+            });
+
+            $.validatorRegistrations[blsPubKeyHash] = ValidatorRegistrationData({
+                registrationSignature: params.registrationSignature,
+                pubkeyG1: params.pubkeyG1,
+                pubkeyG2: params.pubkeyG2,
+                salt: params.salt,
+                expiry: uint64(params.expiry),
+                registeredAt: uint64(block.number)
+            });
+
+            $.validatorIndexes[params.index] = blsPubKeyHash;
+
+            emit ValidatorRegistered({
+                podOwner: address(0),
+                operator: msg.sender,
+                delegateKey: delegateKey,
+                blsPubKeyHash: blsPubKeyHash,
+                validatorIndex: params.index
+            });
+
+            newValidatorCount++;
+        }
+
+        // Update the operator's validator count
+        OperatorData storage operator = $.operators[msg.sender];
+        operator.validatorCount += uint128(newValidatorCount);
+    }
+
+    /**
+     * @inheritdoc IUniFiAVSManager
+     * @dev Restricted in this context is like `whenNotPaused` modifier from Pausable.sol
+     */
+    function verifyValidatorSignatures(bytes32[] calldata blsPubKeyHashes) external restricted {
```

Comment:
I'd change function name. Looking at the name, I assumed that it is supposed to be called with all signature hashes, and that it would validate and mark them as valid.
The function is expected to take in array of 'bad' pubkeys, deregister them & slash the operator.

So maybe use `slashAndDeregisterValidators` and update the natspec in the interface.